emu/src/internal_memory.rs

```diff
@@ -217,4 +232,46 @@ mod tests {
 
         assert_eq!(im.read_at(address), 5);
     }
+
+    #[test]
+    fn write_bg_palette_ram() {
+        let mut im = InternalMemory::new();
+
+        let address = 0x05000008; // WININ lower byte
+
+        im.write_at(address, 10);
+        assert_eq!(im.bg_palette_ram[8], 10);
+    }
+
+    #[test]
+    fn read_bg_palette_ram() {
+        let mut im = InternalMemory::new();
+        im.bg_palette_ram[8] = 15;
+
+        let address = 0x05000008; // WININ lower byte
+        let value = im.read_at(address);
+
+        assert_eq!(value, 15);
+    }
+
+    #[test]
+    fn write_obj_palette_ram() {
+        let mut im = InternalMemory::new();
+
+        let address = 0x05000208; // WININ lower byte
+
+        im.write_at(address, 10);
+        assert_eq!(im.obj_palette_ram[8], 10);
+    }
+
+    #[test]
+    fn read_0bj_palette_ram() {
+        let mut im = InternalMemory::new();
+        im.obj_palette_ram[8] = 15;
+
+        let address = 0x05000208; // WININ lower byte
```

Comment:
Is `// WININ lower byte` right comment here?

Reply:
No, just a misprint copy-paste. I'll remove it.

Reply:
Oh nice :)